Documentation/network/bgp-control-plane/bgp-control-plane-v2.rst

```diff
@@ -736,6 +736,63 @@ Similarly, ``internalTrafficPolicy`` is considered for ``ClusterIP`` advertiseme
     and ignores the configuration of ``.spec.internalTrafficPolicy``.
 
 
+Overlapping Advertisements
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When configuring ``CiliumBGPAdvertisement``, it is possible that two or more
+advertisements match the same Service. Prior to Cilium 1.18, overlapping matches 
+were not expected and the last sequential match was used. Today, overlapping 
+advertisements are supported and their handling varies by attribute:
+
+* Communities: the union of elements is taken across all matches
+* Local Preference: the largest value is selected
+
+As an example, below we have two advertisements which each define a selector 
+match.  One matches on the label ``vpc1`` while the other matches on ``vpc2``.
+
+.. code-block:: yaml
+
+    apiVersion: cilium.io/v2alpha1
+    kind: CiliumBGPAdvertisement
+    metadata:
+      name: bgp-advertisements
+      labels:
+        advertise: bgp
+    spec:
+      advertisements:
+        - advertisementType: "Service"
+          service:
+            addresses:
+              - ClusterIP
+          selector:
+            matchExpressions:
+              - { key: vpc1, operator: In, values: [ true ] }
+          attributes:
+            communities:
+              standard: [ "65000:100" ]
+        - advertisementType: "Service"
+          service:
+            addresses:
+              - ClusterIP
+          selector:
+            matchExpressions:
+              - { key: vpc2, operator: In, values: [ true ] }
+          attributes:
+            communities:
+              standard: [ "65000:200" ]
+
+We have a deployment named `hello-world` which was labeled using:
```

Comment:
Docs linter is complaining here, I think hello-world requires double quotes. 
```
/github/workspace/Documentation/network/bgp-control-plane/bgp-control-plane-v2.rst:784: WARNING: 'any' reference target not found: hello-world

```

Reply:
I will run the test suite after you push a fix for it, and also rebase on latest.

Reply:
Awesome!

I just caught one more issue w/ the docs.  Working on that at the moment.  With a fresh round of local testing, I noticed that the labels such as `vpc1=true` need to be set on the K8s "Service", not the Pod/Deployment.  I missed this before because I was only testing with the more generic:
```
selector:
        matchExpressions:
        - {key: somekey, operator: NotIn, values: ['never-used-value']}
```

I.e., one needs to run:
```
k label service hello-world vpc1=true
k label service hello-world vpc2=true
```

Then, I saw:
```
dswafford-2% k get service hello-world --show-labels
NAME          TYPE           CLUSTER-IP   EXTERNAL-IP   PORT(S)          AGE     LABELS
hello-world   LoadBalancer   10.2.65.71   <pending>     8080:30569/TCP   3m46s   app=hello-world,vpc1=true,vpc2=true
```

And in FRR, I saw:
```
router0# show ip bgp sum

IPv4 Unicast Summary (VRF default):
BGP router identifier 10.0.0.1, local AS number 65000 vrf-id 0
BGP table version 32
RIB entries 1, using 192 bytes of memory
Peers 2, using 1434 KiB of memory
Peer groups 1, using 64 bytes of memory

Neighbor        V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt Desc
fd00:10:0:1::2  4      65001      1532      1441        0    0    0 11:38:59            1        1 N/A
fd00:10:0:2::2  4      65001      1530      1438        0    0    0 11:55:12            1        1 N/A

Total number of neighbors 2
router0# show ip bgp 10.2.65.71
BGP routing table entry for 10.2.65.71/32, version 32
Paths: (2 available, best #1, table default)
  Advertised to non peer-group peers:
  fd00:10:0:1::2 fd00:10:0:2::2
  65001
    fd00:10:0:1::2 from fd00:10:0:1::2 (10.0.1.2)
      Origin IGP, valid, external, multipath, best (Older Path)
      Large Community: 1111:1111:1111 2222:2222:2222
      Last update: Fri Dec 20 18:21:48 2024
  65001
    fd00:10:0:2::2 from fd00:10:0:2::2 (10.0.2.2)
      Origin IGP, valid, external, multipath
      Large Community: 1111:1111:1111 2222:2222:2222
      Last update: Fri Dec 20 18:21:48 2024
```

That was using:
```
apiVersion: cilium.io/v2alpha1
kind: CiliumBGPAdvertisement
metadata:
  name: "bgp-advertisements"
  labels:
    advertise: "bgp"
spec:
  advertisements:
    - advertisementType: "Service"
      service:
        addresses:
        - ClusterIP
        - LoadBalancerIP
      selector:
        matchExpressions:
        - {key: vpc1, operator: In, values: ['true']}
      attributes:
        communities:
          large: [ "1111:1111:1111" ]
    - advertisementType: "Service"
      service:
        addresses:
        - ClusterIP
      selector:
        matchExpressions:
        - {key: vpc2, operator: In, values: ['true']}
      attributes:
        communities:
          large: [ "2222:2222:2222" ]
    - advertisementType: "Service"
      service:
        addresses:
        - ClusterIP
      selector:
        matchExpressions:
        - {key: vpc3, operator: In, values: ['true']}
      attributes:
        communities:
          large: [ "3333:3333:3333" ]
    - advertisementType: "Service"
      service:
        addresses:
        - ClusterIP
      selector:
        matchExpressions:
        - {key: vpc4, operator: In, values: ['true']}
      attributes:
        communities:
          large: [ "4444:4444:4444" ]
---
```

The Containerlab setup here was `contrib/containerlab/bgpv2/service`.